### ISSUE TITLE
[Bug]: Query engine does not use the ServiceContext from the index when `service_context` is omitted from `index.as_query_engine`

### ISSUE THREAD

**Issue by tomtitherington** [jerryjliu/llama_index#6510]
### Bug Description

When the argument `service_context` is omitted from the index constructor `as_query_engine` the query engine uses the default LLMPredictor `text-davinci-003` and not the LLMPredictor defined in the ServiceContext object.

i.e. 

```

index = load_index_from_storage(
    service_context=service_context,
    storage_context=storage_context,
)

# Works
query_engine = index.as_query_engine(
    service_context=service_context, 
    similarity_top_k=3, 
    streaming=True
)  

# DOES NOT work
query_engine = index.as_query_engine()
```

Please see the Steps to Reproduce for the entire example.

### Version

0.0.205

### Steps to Reproduce

```
service_context = ServiceContext.from_defaults(
  llm_predictor=LLMPredictor(
        llm=ChatOpenAI(temperature=0, model_name='gpt-4', streaming=True)
    ),
    chunk_size=512,
  )
)

storage_context = StorageContext.from_defaults(persist_dir="./storage")

index = load_index_from_storage(
    service_context=service_context,
    storage_context=storage_context,
)

# This works
query_engine = index.as_query_engine(
    service_context=service_context, 
    similarity_top_k=3, 
    streaming=True
)  

# This DOES NOT work
query_engine = index.as_query_engine()

response = query_engine.query(query)
print(response)
```

**Comment by Disiok** [comment]
Hmm, this is surprising. Which index is this?

**Comment by Disiok** [comment]
This is the current code for as_query_engine:
```python
    def as_query_engine(self, **kwargs: Any) -> BaseQueryEngine:
        # NOTE: lazy import
        from llama_index.query_engine.retriever_query_engine import RetrieverQueryEngine

        retriever = self.as_retriever(**kwargs)

        kwargs["retriever"] = retriever
        if "service_context" not in kwargs:
            kwargs["service_context"] = self._service_context
        return RetrieverQueryEngine.from_args(**kwargs)
```

**Comment by Disiok** [comment]
Could you say how you build the index? Would help reproducing.

**Comment by tannershamrock** [comment]
I have this same issue. I had to go into the LLMPredictor class and override it with 3.5 Turbo temporarily.  I had about a $100 night the other night, because I didn't realize it was defaulting to Davinci.

**Comment by jma7889** [comment]
I have similar issue, created a bug issue https://github.com/jerryjliu/llama_index/issues/6620 .
I think the way service_context passed around as optional is error prone.

**Comment by jma7889** [comment]
There is a workaround for this bug to use set global context function to override default service context. see https://gpt-index.readthedocs.io/en/latest/guides/primer/usage_pattern.html

**Comment by tyre** [comment]
I have a proposed fix for this in #6911 .

For this issue specifically, while the current `as_query_engine` code does pass in a `service_context`, the initializer for `RetrieverQueryEngine` doesn't actually take one. I fixed that so service context propagation works.

**Comment by logan-markewich** [comment]
@tyre 's fix has merged. Overall though, I would recommend using the global service context to save yourself work in the future

https://gpt-index.readthedocs.io/en/latest/core_modules/supporting_modules/service_context.html#setting-global-configuration
